cmd/cloud-controller-manager/app/controllermanager.go

```diff
@@ -86,13 +86,13 @@ the cloud specific control loops shipped with Kubernetes.`,
 func resyncPeriod(c *cloudcontrollerconfig.CompletedConfig) func() time.Duration {
 	return func() time.Duration {
 		factor := rand.Float64() + 1
-		return time.Duration(float64(c.Generic.ComponentConfig.MinResyncPeriod.Nanoseconds()) * factor)
+		return time.Duration(float64(c.Generic.ComponentConfig.GenericComponent.MinResyncPeriod.Nanoseconds()) * factor)
 	}
 }
 
 // Run runs the ExternalCMServer.  This should never exit.
 func Run(c *cloudcontrollerconfig.CompletedConfig) error {
-	cloud, err := cloudprovider.InitCloudProvider(c.Generic.ComponentConfig.CloudProvider, c.Generic.ComponentConfig.CloudConfigFile)
+	cloud, err := cloudprovider.InitCloudProvider(c.Generic.ComponentConfig.CloudProvider.CloudProvider, c.Generic.ComponentConfig.CloudProvider.CloudConfigFile)
```

Comment:
should `CloudProvider.CloudProvider` be better `CloudProvider.Name` ?

Reply:
fix now